### ISSUE TITLE
Card database won't update/open

### ISSUE THREAD

**Issue by SketchyMofo** [Cockatrice/Cockatrice#4221]
<b>System Information:</b>
Client Version: 2.7.5 (2020-08-23)
Client Operating System: Windows 10 (10.0)
Build Architecture: 64-bit
Qt Version: 5.12.6
System Locale: en_US
Install Mode: Standard
---------------------------------------------------------------------------
Theme changed: "Default"
SoundEngine: disabling sound
Sound theme changed: "Default"
SoundEngine: disabling sound
Language changed: ""
main(): starting main program
main(): MainWindow constructor finished
Spoilers Disabled
main(): ui.show() finished
Startup: found config with current version
CardDatabase::loadCardDatabases start
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/cards.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/tokens.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 0 Sets= 0
CardDatabase::loadCardDatabases failed
PictureLoader: cache fail for "_trice_card_back_243355"
PictureLoader: cache fail for "_trice_card_back_226292"
QCoreApplication::postEvent: Unexpected null receiver
CardDatabase::loadCardDatabases start
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/cards.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/tokens.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 0 Sets= 0
CardDatabase::loadCardDatabases failed
_______________________________________________________________________________________

A few days ago, I tried to open Cockatrice and it asked me to update the card database. I chose yes and after trying to load the database from Oracle, it keeps giving me this message and asking me to upload the card database again. This sends it into a loop of these actions.

![CockError](https://user-images.githubusercontent.com/77674064/105034966-4bb0c600-5a28-11eb-9bfb-5734c4c9af5c.PNG)

![Capture](https://user-images.githubusercontent.com/77674064/105035333-d09bdf80-5a28-11eb-9e48-767fb5f57eba.PNG)

I uninstalled Cockatrice and deleted all the files, redownloading everything to my computer. Now the card database won't even load, doing the same thing as before. Has anyone else had this issue before? How do I fix this?

![NoCard](https://user-images.githubusercontent.com/77674064/105035192-97fc0600-5a28-11eb-9563-87fa3d1564b4.PNG)

**Comment by ebbit1q** [comment]
have you tried if the beta works?

**Comment by SketchyMofo** [comment]
> have you tried if the beta works?

I don't think I have. Where do I download that?

**Comment by ebbit1q** [comment]
you can select the beta channel in the settings or get it from here directly https://github.com/Cockatrice/Cockatrice/releases

**Comment by SketchyMofo** [comment]
> you can select the beta channel in the settings or get it from here directly https://github.com/Cockatrice/Cockatrice/releases

I just tried to download the beta and got the same results as when I try to download the database for the original version.

**Comment by ebbit1q** [comment]
I'm unable to reproduce the issue, could you also give the log for the beta?

**Comment by SketchyMofo** [comment]
> I'm unable to reproduce the issue, could you also give the log for the beta?

Sure.

Client Version: 2.7.6-beta.2 (2020-11-23)
Client Operating System: Windows 10 (10.0)
Build Architecture: 64-bit
Qt Version: 5.12.6
System Locale: en_US
Install Mode: Standard
---------------------------------------------------------------------------
Theme changed: "Default"
SoundEngine: disabling sound
Sound theme changed: "Default"
SoundEngine: disabling sound
Language changed: ""
main(): starting main program
main(): MainWindow constructor finished
Spoilers Disabled
main(): ui.show() finished
Startup: found config with current version
CardDatabase::loadCardDatabases start
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/cards.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/tokens.xml" Status = 4 Cards = 0 Sets= 0
[CardDatabase] loadCardDatabase(): Path = "C:/Users/Matt/AppData/Local/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 0 Sets= 0
CardDatabase::loadCardDatabases failed
QWindowsWindow::setGeometry: Unable to set geometry 1366x706+0+23 (frame: 1382x745-8-8) on QWidgetWindow/"MainWindowClassWindow" on "\\.\DISPLAY1". Resulting geometry: 1366x705+0+23 (frame: 1382x744-8-8) margins: 8, 31, 8, 8 minimum size: 806x706 MINMAXINFO maxSize=0,0 maxpos=0,0 mintrack=822,745 maxtrack=0,0)
PictureLoader: cache fail for "_trice_card_back_226292"
PictureLoader: cache fail for "_trice_card_back_226290"

**Comment by ZeldaZach** [comment]
Have you had any luck on this @SketchyMofo ?

**Comment by SketchyMofo** [comment]
> Have you had any luck on this @SketchyMofo ?

So I think I had figured out what was wrong. I was missing a couple of files in the Cockatrice appdata and whenever I tried to re-download Cockatrice, they wouldn't restore. They were the cards and download files.

I never figured out how to fix this but I got a solution. A buddy of mine has Cockatrice and he just DMed a copy of his program. I replaced the decks and logged into my account. It's pretty much worked for me for the past couple of weeks. I still don't know what happened to my original game though.
